python/ray/ml/examples/pytorch/permuted_mnist_domain_incremental_learning_example.py

```diff
@@ -0,0 +1,297 @@
+            train_mb_y = train_mb_y.to(train.torch.get_device())
+
+            # Forward
+            logits = model(train_mb_x)
+            # Loss
+            loss = criterion(logits, train_mb_y)
+            # Backward
+            loss.backward()
+            # Update
+            optimizer.step()
+
+            if train.world_rank() == 0 and iteration % 100 == 0:
+                print(f"loss: {loss.item():>7f}, iteration: {iteration}")
+
+        # Checkpoint model after every epoch.
+        train.save_checkpoint(model=model.module.state_dict())
+
+
+def preprocess_images(df: pd.DataFrame) -> pd.DataFrame:
+    """Preprocess images by scaling each channel in the image."""
+
+    torchvision_transforms = transforms.Compose(
+        [ToTensor(), transforms.Normalize((0.1307,), (0.3081,))]
+    )
+
+    new_df = df.copy()
+    new_df["image"] = df["image"].map(torchvision_transforms)
+    return new_df
+
+
+def incremental_train(dataset_stream: Iterator[ray.data.Dataset]) -> Checkpoint:
+    """Incrementally trains MLP model on each task in a distributed fashion.
+
+    Args:
+        dataset_stream: The stream of datasets to train on.
+    """
+
+    latest_checkpoint = None
+
+    for dataset in dataset_stream:
+        trainer = TorchTrainer(
+            train_loop_per_worker=train_loop_per_worker,
+            train_loop_config={
+                "num_epochs": 2,
+                "learning_rate": 0.001,
+                "momentum": 0.9,
+                "batch_size": 32,
+            },
+            scaling_config={"num_workers": 2, "use_gpu": False},
+            datasets={"train": dataset},
+            preprocessor=BatchMapper(fn=preprocess_images),
+            resume_from_checkpoint=latest_checkpoint,
+        )
+        result = trainer.fit()
+        latest_checkpoint = result.checkpoint
+
+    return latest_checkpoint
+
+
+def predict(checkpoint: Checkpoint, test_dataset_stream: Iterator[ray.data.Dataset]):
+    """Runs prediction of the incrementally trained model on stream of test datasets.
+
+    Args:
+        checkpoint: The checkpoint containing the trained model.
+        test_dataset_stream: A stream of Ray Datasets to test on.
+    """
+
+    acc_results = []
+    batch_predictor = BatchPredictor.from_checkpoint(
+        checkpoint=checkpoint,
+        predictor_cls=TorchPredictor,
+        model=SimpleMLP(num_classes=10),
+    )
+
+    for test_dataset in test_dataset_stream:
+        model_output = batch_predictor.predict(
+            data=test_dataset, feature_columns=["image"], unsqueeze=False
+        )
+        # Postprocess model outputs.
+        # Convert logits outputted from model into actual class predictions.
+        # TODO: Find a better way to do this. This is pretty ugly right now.
+        # TODO: See if Predictors can accept "postprocessing" transforms to be applied
+        #  directly after model output.
+
+        def convert_logits_to_classes(df):
```

Comment:
this seems like a super common operation.
in torch, can you return both the logits and argmax(logits) from the forward function? basically, make argmax part of the model.